### PR TITLE
Feature/ui new log viewer partially working slider

### PR DESCRIPTION
**Pull request by PGuimarais**
Top Slider:
- Added top slider
- Sliding enabled; captures point in time across the timeline

Timeline:
- 'Warn' and 'Error' events render properly and stack if they occur at same point along x-axis 
- Clicking and dragging along the timeline allows for selecting time window

### CODE REVIEW

File: cdap-ui/app/directives/timeline/timeline.js [hunk 1]
```diff
@@ -154,44 +166,140 @@ function link (scope, element) {
...
+                  } else {
+                    return 'otherCircle';
                   }
                 });
 
+        //Brush
+        var brush = d3.svg.brush()
+            .x(xScale)
+            .extent([0,0])
+            .on('brush', brush);
```
Comment (elfenheart) [inline cdap-ui/app/directives/timeline/timeline.js]: I think this will cause recursive events

File: cdap-ui/app/directives/timeline/timeline.js [hunk 2]
```diff
@@ -154,44 +166,140 @@ function link (scope, element) {
...
-                .attr('cy', height-height/3)
-                .attr('r', 3)
+                .attr('cy', function(d) {
+                  let numDots = timelineStack[Math.floor(xScale(d.time))]--;
+                  return height-height/3 - (numDots * 6);
+                })
+                .attr('r', 2)
                 .attr('class', function(d) {
                   if(d.level === 'ERROR'){
                     return 'redCircle';
```
Comment (elfenheart) [inline cdap-ui/app/directives/timeline/timeline.js]: css class should be dash-separated

File: cdap-ui/app/directives/timeline/timeline.js [hunk 3]
```diff
@@ -154,44 +166,140 @@ function link (scope, element) {
...
+        var brush2 = d3.svg.brush()
+            .x(xScale)
+            .extent([0,0])
+            .on('brush', slidePin);
+
+        var svg2 = d3.select('.top-bar').append('svg')
+            .attr('width', width)
+            .attr('height', 15)
+          .append('g')
+            .attr('transform' , 'translate(0,' + 0 + ')');
```
Comment (elfenheart) [inline cdap-ui/app/directives/timeline/timeline.js]: this won't be doing anything right?

File: cdap-ui/app/directives/timeline/timeline.js [hunk 4]
```diff
@@ -124,82 +143,218 @@ function link (scope, element) {
...
-        var xAxis = d3.svg.axis()
-          .orient('bottom')
-          .scale(xScale);
-
-        circles.attr('cx', function(d, i) {
-                  return i*50 + 25;
-                  // console.log("Time is: " + d.time);
-                  // return xScale(d.time);
+        circles.attr('cx', function(d) {
+                  let xVal = Math.floor(xScale(d.time));
```
Comment (ajainarayanan) [inline cdap-ui/app/directives/timeline/timeline.js]: formatting/indentation is way off here I think?

**Comment by ajainarayanan** [comment]
@PGuimarais I understand that this is an initial implementation of timeline directive but some of the parameters that we have used inside the directive should be configurable (like the most important one being data) which should be passed in an isolate scope. Not necessarily in this PR but in the next one lets try to make it a little bit more configurable so that it becomes and drop-and-use dumb directive (given the data).

**Comment by PGuimarais** [comment]
@ajainarayanan sounds good

**Comment by ajainarayanan** [comment]
Latest Bamboo build: http://builds.cask.co/browse/CDAP-DRC3907-8

**Comment by ajainarayanan** [comment]
screen shot 2016-06-29 at 7 49 57 pm

@PGuimarais I think we have a mis-communication here. The white anchor on top of the timeline is immutable. The timeline along with the white anchor should be fixed on the screen and when the user scrolls on the logs table the white anchor moves above the timeline indicating the time of the logs the user is scrolling in the table.

**Comment by ajainarayanan** [comment]
screen shot 2016-06-29 at 7 53 31 pm

It would be nicer if we can click on the text & have the checkbox selected.

File: cdap-ui/app/directives/log-viewer/log-viewer.less [hunk 5]
```diff
@@ -31,9 +31,63 @@
 my-log-viewer {
   .log-viewer-container {
 
+    .slider .background {
+      cursor: pointer !important;
```
Comment (elfenheart) [inline cdap-ui/app/directives/log-viewer/log-viewer.less]: don't use `!important`

File: cdap-ui/app/directives/log-viewer/log-viewer.less [hunk 6]
```diff
@@ -31,9 +31,63 @@
...
+    }
+
+    .fill-bar {
+      fill: green;
+      stroke: #171717;
+      stroke-opacity: 0.75;
+      stroke-width: 100%;
+    }
+
+    .leftHandle {
```
Comment (elfenheart) [inline cdap-ui/app/directives/log-viewer/log-viewer.less]: css class should be dash-separated

File: cdap-ui/app/directives/timeline/timeline.js [hunk 7]
```diff
@@ -119,73 +135,250 @@ function link (scope, element) {
...
-      let height = 60;
+      var width = element.parent()[0].offsetWidth;
+      var height = 50;
+      var paddingLeft = 15;
+      var paddingRight = 15;
+      var maxRange = width - paddingLeft - paddingRight;
+      var sliderLimit = maxRange + 24;
+      var pinX = width-8;
+      var sliderX = 0;
+      var timelineStack = {};
```
Comment (elfenheart) [inline cdap-ui/app/directives/timeline/timeline.js]: use `let`

File: cdap-ui/app/directives/timeline/timeline.less [hunk 8]
```diff
@@ -16,6 +16,26 @@
 
 my-timeline {
   .timeline-container {
+    color: white;
+
+    .scrollPin {
```
Comment (elfenheart) [inline cdap-ui/app/directives/timeline/timeline.less]: class should be dash-delimited

File: cdap-ui/app/directives/timeline/timeline.less [hunk 9]
```diff
@@ -16,6 +16,26 @@
...
+      cursor: pointer;
+      fill: #fefefe;
+    }
+
+    .scrollNeedle {
+      cursor: pointer;
+      fill: blue;
+      stroke: #2397e8;
```
Comment (elfenheart) [inline cdap-ui/app/directives/timeline/timeline.less]: since this color is being reused, extract this to a less variable

**Comment by elfenheart** [comment]
needs to be rebased

**Comment by PGuimarais** [comment]
Will reopen against develop

**Pull request closed without merging**